Cyborg/SwiftUIView.swift

```diff
@@ -0,0 +1,143 @@
+//
+//  http://www.apache.org/licenses/LICENSE-2.0
+//
+//  Unless required by applicable law or agreed to in writing, software
+//  distributed under the License is distributed on an "AS IS" BASIS,
+//  WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+//  See the License for the specific language governing permissions and
+//  limitations under the License.
+//  
+
+#if canImport(SwiftUI)
+
+import SwiftUI
+
+/// A `VectorDrawableView` for SwiftUI.
+///
+/// # Experimental
+/// This feature is experimental and may change.
+/// Current areas where this API may be deficient are:
+///
+/// 1. The drawable must be fully parsed before the view's init is called, which isn't in the spirit of SwiftUI's views being
+/// cheap to construct.
+///
+/// 2. Handling of resources is a bit hacky. See below for details.
+///
+/// 3. Further testing of how the view handles sizing is required.
+///
+/// # Usage
+///
+/// You initialize a `VectorDrawableView` with
+/// `init(_ drawable:)`.
+///
+/// You can provide the `theme` and `resources` through
+/// `environment()`. It seems as though `theme` is better suited to be an `EnvironmentObject` than
+/// merely part of the `Environment`, but there doesn't seem to be any way to access an `EnvironmentObject` inside of
+/// `updateUIView` without crashing. If you want dynamic theme changes, the easiest way currently is to bind an `EnvironmentObject`
+/// higher in your view hierarchy, then use its output to mutate the `Environment` by calling `View.environment` on the child
+/// tree that contains the drawables you want to theme.
+///
+/// - Note: `resources` does not update after the view is first created,
+/// it's passed as an `Environment` var solely for convenience.
+@available(iOS 13.0, *)
+public struct VectorDrawableView: UIViewRepresentable {
+    
+    public let drawable: VectorDrawable
+    
+    /// Initializer.
+    public init(_ drawable: VectorDrawable) {
+        self.drawable = drawable
+    }
+    
+    public func makeUIView(context: UIViewRepresentableContext<VectorDrawableView>) -> VectorView {
+        let view = VectorView(theme: context.environment.vectorDrawableTheme,
+                              resources: context.environment.vectorDrawableResources)
+        print(context.environment.vectorDrawableTheme.colorFromTheme(named: ""))
+        view.setContentHuggingPriority(.defaultHigh,
+                                       for: .horizontal)
+        view.setContentHuggingPriority(.defaultHigh,
+                                       for: .vertical)
+        view.drawable = drawable
+        return view
+    }
+    
+    public func updateUIView(_ uiView: VectorView,
+                             context: UIViewRepresentableContext<VectorDrawableView>) {
+        uiView.drawable = drawable
+        uiView.theme = context.environment.vectorDrawableTheme
+    }
+
+}
+
+/// The key for Vector Drawable themes.
+public struct ThemeKey: EnvironmentKey {
```

Comment:
Having watch a couple more WWDC videos on SwiftUI do think it is worthwhile to create your own Theme environment variable? Could we use the `\.colorScheme` key and `accentColor` modifier instead? Or I guess add our own `accentColor` method to make it feel like a first class citizen?

Reply:
How would that work? 

`ColorScheme` limits us in terms of what colors we can read (I don't think we can even get system colors from it). 

`accentColor` only seems to work as a tint, and we need to support arbitrary colors.

I agree that this is kind of bad, but I don't see any other options.

Reply:
In my mind to be a first class citizen of SwiftUI modifiers like `foregroundColor`, `tintColor`, etc should work for a custom view. Requiring custom environment values for a single custom view seems heavy handed. If you had half a dozen custom views that all required unique environment variables to style them I can't see that being enjoyable. Also if we wanted to create another custom view in a different open source repository that also had a theme we'd have to make these repos dependent on one another which isn't ideal.

If there is no way know if a modifier has been applied to your custom view I would think that implementing the modifiers would be the next step. If a modifier is called you then store the values locally and use it to update the look of your custom view.

With respect to dark mode I will admit I don't have first hand experience but I read that Apple allows you to define a color set for normal mode as well as dark mode and that we would expect developers to leverage this and our custom views should also work in this system and not require developers to do any extra work.

Reply:
I don't think you can get the environment values out of SwiftUI in a form that you can put into UIKit. If you print the contents of `EnvironmentValues`, there's a bunch of private types that are used as keys.

Reply:
Boooooo what kind of 1.0 is this 🍎?